test/Microsoft.AspNetCore.Server.KestrelTests/FrameTests.cs

```diff
@@ -531,17 +565,47 @@ public void RequestAbortedTokenIsResetBeforeLastWriteWithContentLength()
             Assert.NotSame(original, _frame.RequestAborted.WaitHandle);
         }
 
+        [Fact]
+        public async Task ExceptionDetailNotIncludedWhenLogLevelInformationNotEnabled()
+        {
+            var previousLog = _serviceContext.Log;
+
+            try
+            {
+                var mockTrace = new Mock<IKestrelTrace>();
+                mockTrace
+                    .Setup(trace => trace.IsEnabled(LogLevel.Information))
+                    .Returns(false);
+
+                _serviceContext.Log = mockTrace.Object;
+
+                await _input.Writer.WriteAsync(Encoding.ASCII.GetBytes($"GET /%00 HTTP/1.1\r\n"));
+                var readableBuffer = (await _input.Reader.ReadAsync()).Buffer;
+
+                var exception = Assert.Throws<BadHttpRequestException>(() =>
+                    _frame.TakeStartLine(readableBuffer, out _consumed, out _examined));
+                _input.Reader.Advance(_consumed, _examined);
+
+                Assert.Equal("Invalid request target: ''", exception.Message);
```

Comment:
~~Should this be `"Invalid request target: '/\x00'"`?~~

Nvm, same as below. Can you add a comment that this is correct even though it looks wrong at first-glance